warehouse/transformer/transformer_fuzz_test.go

```diff
@@ -0,0 +1,668 @@
+			f.Add(destType, `{}`, `{"type":"extract","recordId":"recordID","event":"users","receivedAt":"`+dateFormat+`","channel":"web","properties":{"name":"Home","title":"Home | RudderStack","url":"https://www.rudderstack.com"},"context":{"traits":{"name":"Richard Hendricks","email":"rhedricks@example.com","logins":2},"ip":"1.2.3.4"}}`)
+		}
+	}
+
+	f.Fuzz(func(t *testing.T, destType, destConfigJSON, payload string) {
+		if _, exist := whutils.WarehouseDestinationMap[destType]; !exist {
+			return
+		}
+
+		var destConfig map[string]any
+		err = json.Unmarshal([]byte(destConfigJSON), &destConfig)
+		if err != nil {
+			return
+		}
+
+		sanitizedPayload, err := sanitizePayload(payload)
+		if err != nil {
+			if !errors.Is(err, errInvalidDateTime) {
+				return
+			}
+
+			t.Log("Destination Type: ", destType, "Destination Config: ", destConfigJSON, "Payload: ", payload)
+			t.Skip()
+		}
+
+		var (
+			eventType  = gjson.Get(sanitizedPayload, "type").String()
+			eventName  = gjson.Get(sanitizedPayload, "event").String()
+			messageID  = gjson.Get(sanitizedPayload, "messageId").String()
+			receivedAt = gjson.Get(sanitizedPayload, "receivedAt").Time()
+			recordID   = gjson.Get(sanitizedPayload, "recordId").Value()
+		)
+
+		if len(messageID) == 0 || receivedAt.IsZero() {
+			return
+		}
+
+		sanitizedPayload, err = sjson.Set(sanitizedPayload, "receivedAt", receivedAt.Format(misc.RFC3339Milli))
+		if err != nil {
+			return
+		}
+
+		conf := setupConfig(transformerResource, map[string]any{})
+
+		processorTransformer := ptrans.NewTransformer(conf, logger.NOP, stats.Default)
+		warehouseTransformer := New(conf, logger.NOP, stats.NOP)
+
+		eventContexts := []testhelper.EventContext{
+			{
+				Payload: []byte(sanitizedPayload),
+				Metadata: ptrans.Metadata{
+					EventType:       eventType,
+					EventName:       eventName,
+					DestinationType: destType,
+					ReceivedAt:      receivedAt.Format(misc.RFC3339Milli),
+					SourceID:        "sourceID",
+					DestinationID:   "destinationID",
+					SourceType:      "sourceType",
+					MessageID:       messageID,
+					RecordID:        recordID,
+				},
+				Destination: backendconfig.DestinationT{
+					Name:   destType,
+					Config: destConfig,
+					DestinationDefinition: backendconfig.DestinationDefinitionT{
+						Name: destType,
+					},
+				},
+			},
+		}
+		cmpEvents(t, eventContexts, processorTransformer, warehouseTransformer)
+	})
+}
+
+func cmpEvents(t *testing.T, eventContexts []testhelper.EventContext, pTransformer, dTransformer ptrans.DestinationTransformer) {
```

Comment:
Can we use `ValidateEvents` method from the `testhelper` package ?

Reply:
No, we can't use ValidateEvents. It's basically for comparing the processor and destination transformer response with the expected response.